notebooks/2018-12-14_Fortran_Files.ipynb

```diff
@@ -0,0 +1,333 @@
+    "\n",
+    "There are some good practices on how to deal with files in Fortran. \n",
+    "These make it easier to port data, because they will work on different systems.\n",
+    "\n",
+    "### Ensure that you know the kind of the variable\n",
+    "\n",
+    "If you write something like\n",
+    "\n",
+    "```fortran\n",
+    "integer :: ii\n",
+    "```\n",
+    "\n",
+    "you don't really know what kind of integer `ii` will be. \n",
+    "Often, you can set the default integer and real kind with compiler options, but it's far better to explicitly declare the kind in the code itself.\n",
+    "\n",
+    "Since Fortran 2003 -- and all compilers we use today are compatible with this -- you can use the intrinsic `iso_fortran_env` module to get the proper kinds:\n",
+    "\n",
+    "```fortran\n",
+    "use iso_fortran_env, only: int32, real64\n",
+    "implicit none\n",
+    "integer(kind=int32) :: ii\n",
+    "real(kind=real64) :: x(10, 100)\n",
+    "```\n",
+    "\n",
+    "In old code, you might find statements like:\n",
+    "\n",
+    "```fortran\n",
+    "integer*4 ii        ! DO NOT DO THIS\n",
+    "```\n",
+    "\n",
+    "This syntax has *never* been standard, and I strongly discourage you from using it.\n",
+    "Slightly better, but still wrong, is this:\n",
+    "\n",
+    "```fortran\n",
+    "integer(kind=4) :: ii   ! Still not good\n",
+    "```\n",
+    "\n",
+    "There is no guarantee that every compiler will use the same kind values for the same variable types.\n",
+    "If for some reason you can not use `iso_fortran_env`, use the `selected_int_kind` and `selected_real_kind` methods instead:\n",
+    "\n",
+    "```fortran\n",
+    "integer, parameter :: real64 = selected_real_kind(15, 307)\n",
+    "real(kind=real64) :: x(10, 100)\n",
+    "```\n",
+    "\n",
+    "See the table below for which type you need\n",
+    "\n",
+    "| bytes | int name | integer kind | integer max |   |  | real name | real kind |\n",
+    "|------|-------|------|------|-----|    |-------|-------|-------|\n",
+    "|  2  | `int16` | `selected_int_kind(3)` |  127 | | | |\n",
+    "|  4  | `int32` | `selected_int_kind(5)` | > 2*10^9 | | | `real32` | `selected_real_kind(6, 37)` |\n",
+    "|  8  | `int64` | `selected_int_kind(10)` | > 9*10^18 | | | `real64` | `selected_real_kind(15, 307)` |\n",
+    "| 16  | ----    | `selected_int_kind(19)` | > 10^38 | | | `real128` | `selected_real_kind(33, 4931)` |\n",
+    "\n",
+    "Note that `iso_fortran_env` does not have a named type `int128`, though your compiler might have it. \n",
+    "Some compilers also have a 10-byte real kind.\n",
+    "\n",
+    "### newunit\n",
+    "\n",
+    "Whenever you interact with a file, you need a unit, an integer value that references a specific open file.\n",
+    "Some I/O streams, specifically Standard Input, Standard Output, and Standard Error have compiler dependent values for these units, which unfortunately are not standardised.\n",
```

Comment:
Not sure why you're talking about stdout etc here, but they also have defined constants in `iso_fortran_env`